### ISSUE TITLE
Support Java 24-ea

### ISSUE THREAD

**Issue by garydgregory** [easymock/easymock#673]
Hi All,

Over at Apache Commons, we are building with LTS and EA versions of Java. 

It looks like Java 24-EA is not supported yet. For example:
```
[INFO] Running org.apache.commons.collections4.IteratorUtilsTest
Error:  Tests run: 49, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.690 s <<< FAILURE! -- in org.apache.commons.collections4.IteratorUtilsTest
Error:  org.apache.commons.collections4.IteratorUtilsTest.testGetIterator -- Time elapsed: 0.563 s <<< ERROR!
java.lang.IllegalArgumentException: Could not create type
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:170)
	at net.bytebuddy.TypeCache$WithInlineExpunction.findOrInsert(TypeCache.java:399)
	at org.easymock.internal.ClassProxyFactory.doCreateProxy(ClassProxyFactory.java:171)
	at org.easymock.internal.ClassProxyFactory.createProxy(ClassProxyFactory.java:154)
	at org.easymock.internal.MocksControl.createMock(MocksControl.java:110)
	at org.easymock.internal.MocksControl.createMock(MocksControl.java:83)
	at org.easymock.IMocksControl.mock(IMocksControl.java:44)
	at org.easymock.EasyMock.mock(EasyMock.java:70)
	at org.easymock.EasyMock.createMock(EasyMock.java:322)
	at org.apache.commons.collections4.IteratorUtilsTest.testGetIterator(IteratorUtilsTest.java:790)
	at java.base/java.lang.reflect.Method.invoke(Method.java:573)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
Caused by: java.lang.IllegalArgumentException: java.util.Dictionary$$$EasyMock$2 must be defined in the same package as org.easymock.internal.ClassProxyFactory
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingLookup.injectRaw(ClassInjector.java:1635)
	at net.bytebuddy.dynamic.loading.ClassInjector$AbstractBase.inject(ClassInjector.java:118)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$UsingLookup.load(ClassLoadingStrategy.java:519)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:101)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:6325)
	at org.easymock.internal.ClassProxyFactory.lambda$doCreateProxy$0(ClassProxyFactory.java:181)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:168)
	... 12 more
```
The above is from https://github.com/apache/commons-collections/actions/runs/11265666158/job/31327872473
Using:
```
Apache Maven 3.8.8 (4c87b05d9aedce574290d1acc98575ed5eb6cd39)
Maven home: /usr/share/apache-maven-3.8.8
Java version: 24-beta, vendor: Eclipse Adoptium, runtime: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/24.0.0-ea.18.0.ea/x64
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "6.5.0-1025-azure", arch: "amd64", family: "unix"
```

TY

**Comment by garydgregory** [comment]
Hello @henri-tremblay 
Thank you for the work. Did you happen to push a SNAPSHOT build somewhere or do I need to build and install locally?

**Comment by henri-tremblay** [comment]
No. Please build locally.

**Comment by garydgregory** [comment]
Hi @henri-tremblay 
I ended up building and installing locally and the test failure is gone. TY!
I'll update our POM as soon as you make a release available.
TY again for the prompt turnaround.

**Comment by henri-tremblay** [comment]
Fixed by 9f361f797322f2c6242ec5bc4c6ef2765ff34276
